tools/java/common/src/com/google/i18n/phonenumbers/BuildMetadataFromXml.java

```diff
@@ -408,36 +419,45 @@ static boolean isValidNumberType(String numberType) {
    * @return  complete description of that phone number type
    */
   // @VisibleForTesting
-  static PhoneNumberDesc.Builder processPhoneNumberDescElement(PhoneNumberDesc.Builder generalDesc,
-                                                               Element countryElement,
-                                                               String numberType,
-                                                               boolean liteBuild) {
+  static PhoneNumberDesc processPhoneNumberDescElement(PhoneNumberDesc generalDesc,
+                                                       Element countryElement,
+                                                       String numberType,
+                                                       boolean liteBuild) {
     NodeList phoneNumberDescList = countryElement.getElementsByTagName(numberType);
-    PhoneNumberDesc.Builder numberDesc = PhoneNumberDesc.newBuilder();
+    PhoneNumberDesc numberDesc = new PhoneNumberDesc();
     if (phoneNumberDescList.getLength() == 0 && !isValidNumberType(numberType)) {
-      numberDesc.setNationalNumberPattern("NA");
-      numberDesc.setPossibleNumberPattern("NA");
+      numberDesc.nationalNumberPattern = "NA";
+      numberDesc.possibleNumberPattern = "NA";
       return numberDesc;
     }
-    numberDesc.mergeFrom(generalDesc.build());
+    if (!generalDesc.nationalNumberPattern.equals("")) {
```

Comment:
Move to new helper method ```copyGeneralDesc(...)```.
Also, why don't you copy when empty? It shouldn't matter.